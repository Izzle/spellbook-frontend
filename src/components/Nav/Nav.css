.Nav__list {
    list-style-type: none;
    display: flex;
    justify-content: space-around;
    align-items: center;
}