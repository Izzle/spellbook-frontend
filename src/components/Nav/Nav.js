import React, { Component} from 'react';
import { Link } from 'react-router-dom';
import './Nav.css';

export default class Nav extends Component {

    render() {
        // The Nav should render as a Hamburge icon on mobile and show the items on a larger screen
        return <ul className='Nav__list'>
            <li><Link to='/' className='Nav__Link'>Home</Link></li>
            <li><Link to='/spells' className='Nav__Link'>Spell Library</Link></li>
            <li><Link to='/spell-creator' className='Nav__Link'>Create a Spell</Link></li>
            <li><Link to='/spellbook' className='Nav__Link'>View Spellbooks</Link></li>
        </ul>
    }
}