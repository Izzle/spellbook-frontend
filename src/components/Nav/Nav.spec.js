import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Nav from './Nav';
import { BrowserRouter } from 'react-router-dom';


describe('<Nav />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter><Nav /></BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        const tree = renderer.create(<BrowserRouter><Nav /></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});