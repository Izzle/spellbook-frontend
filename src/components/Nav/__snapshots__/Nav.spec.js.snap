// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders this UI as expected 1`] = `
<ul
  className="Nav__list"
>
  <li>
    <a
      className="Nav__Link"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </li>
  <li>
    <a
      className="Nav__Link"
      href="/spellbook"
      onClick={[Function]}
    >
      View Spellbooks
    </a>
  </li>
  <li>
    <a
      className="Nav__Link"
      href="/spell-creator"
      onClick={[Function]}
    >
      Create a Spell
    </a>
  </li>
  <li>
    <a
      className="Nav__Link"
      href="/spells"
      onClick={[Function]}
    >
      Spells
    </a>
  </li>
</ul>
`;
